packages/plugin-ext/src/main/browser/webview/webview.ts

```diff
@@ -0,0 +1,212 @@
+/********************************************************************************
+ * Copyright (C) 2018 Red Hat, Inc. and others.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the Eclipse
+ * Public License v. 2.0 are satisfied: GNU General Public License, version 2
+ * with the GNU Classpath Exception which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ ********************************************************************************/
+import { BaseWidget } from '@theia/core/lib/browser/widgets/widget';
+import { IdGenerator } from '../../../common/id-generator';
+import { MiniBrowser } from '@theia/mini-browser/lib/browser/mini-browser';
+import { Disposable, DisposableCollection } from '@theia/core';
+
+export interface WebviewWidgetOptions {
+    readonly allowScripts?: boolean;
+}
+
+export interface WebviewEvents {
+    onMessage?(message: any): void;
+    onKeyboardEvent?(e: KeyboardEvent): void;
+    onLoad?(contentDocument: Document): void;
+}
+
+export class WebviewWidget extends BaseWidget {
+    private static readonly ID = new IdGenerator('webview-widget-');
+
+    protected readonly toDispose = new DisposableCollection();
+    private iframe: HTMLIFrameElement;
+    private state: string | undefined = undefined;
+    private loadTimeout: number | undefined;
+    //    private pendingMessages
```

Comment:
This class has a lot of commented lines, can you remove them?

Reply:
Ok